### ISSUE TITLE
JSR223: Include automationManager in the Default preset

### ISSUE THREAD

**Issue by dilyanpalauzov** [openhab/openhab-core#4368]
When writing rules in JSR 223, one ultimately has to access `automationManager` to do something useful.  The presets - Default, RuleSimple, RuleSupport, RuleFactories, ScriptAction and cache - provide some implicit `import`s and objects (variables).  The latter objects enable connecting the JSR223 code with the OH internals and installing some meaningful automation.

As the only way described at https://www.openhab.org/docs/configuration/jsr223.html to install a rule is to use `automationManager` and thus to import the RuleSupport preset, this preset has always to be included explicitly by the caller.  The other items from the preset are not strictly necessary, as they can be obtained by `import`s. (And the imports are required to move as much code in Groovy as possible under the TypeChecked and CompileStatic annotations).

**Comment by jimtng** [comment]
You can obtain `automationManager` through `scriptExtension.get("automationManager")`. `scriptExtension` is [injected by default](https://github.com/openhab/openhab-core/blob/72753be8cc9af029c3e656a89b29064f4624c73f/bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/ScriptEngineManagerImpl.java#L129), without having to import _any_ presets.

**Comment by dilyanpalauzov** [comment]
Yes, this is indeed the case.  But this is not documented at https://www.openhab.org/docs/configuration/jsr223.html, so `scriptExtension.importPreset("RuleSupport")` is always called in order to get access to `automationManager`.

**Comment by lolodomo** [comment]
You are welcome to improve the existing documentation.

**Comment by dilyanpalauzov** [comment]
> You are welcome to improve the existing documentation.

When I try to improve the existing documentation, changes can get stuck: https://github.com/openhab/openhab-docs/pulls/dilyanpalauzov

**Comment by lolodomo** [comment]
Try to ping @florian-h05

**Comment by florian-h05** [comment]
I don't think automationManager belongs into the Default preset, as it is related to supporting rule creation.
The possibility to statically import other members of the RuleSupport preset should not change that relation.

I would suggest to improve the docs as @jimtng suggested and ping me, so I can take care the PR gets reviewed and merged.

**Comment by dilyanpalauzov** [comment]
> I don't think automationManager belongs into the Default preset, as it is related to supporting rule creation.

Yes, if no rule creation is going to happen, why will somebody in first place use the JSR223 extension mechanism?  JSR223 is used to create and activate rules.

**Comment by lolodomo** [comment]
Looks like this "issue" can be closed.
